tutorials/getting-started-netcore.rst

```diff
@@ -0,0 +1,186 @@
+
+.. code-block:: shell
+
+    dotnet new --install Freya.Template::*
+
+This process will take a few minutes to install the templates on your host,
+after which, you can create a Freya project by simply typing
+
+.. code-block:: shell
+
+    dotnet new freya
+
+.. hint::
+
+    By default, the Freya template will use the ``Async`` construct from ``FSharp.Core``. If you want to use the ``Job`` construct from `Hopac <https://hopac.github.io/Hopac/Hopac.html>`_, then create the project using ``dotnet new freya -f hopac``.
+
+This will create a basic Freya project for you, including four files. :file:`Api.fs`,
+:file:`KestrelInterop.fs`, :file:`Program.fs`, and a :file:`{Freya}.fsproj` file.
+
+First, let's take a look at :file:`Api.fs`. Here's the complete code,
+before the implementation is broken down:
+
+.. code-block:: fsharp
+
+    module Api
+
+    open Freya.Core
+    open Freya.Machines.Http
+    open Freya.Types.Http
+    open Freya.Routers.Uri.Template
+
+    let routeName_ = Route.atom_ "name"
+
+    let name = freya {
+      let! nameO = Freya.Optic.get routeName_
+
+      match nameO with
+      | Some name -> return name
+      | None -> return "World"
+    }
+
+    let sayHello = freya {
+      let! name = name
+
+      return Represent.text (sprintf "Hello, %s!" name)
+    }
+
+    let helloMachine = freyaMachine {
+      methods [GET; HEAD; OPTIONS]
+
+      handleOk sayHello
+    }
+
+    let root = freyaRouter {
+      resource "/hello{/name}" helloMachine
+    }
+
+Dependencies
+------------
+
+On .NET Core, the primary server supported by Freya is Kestrel, but any OWIN-compatible server can be used. The default template already pulls in the necessary dependencies on Kestrel and its OWIN bindings, so you don't need to change anything there.
+
+Freya is distributed as a meta-package. The project file only includes a reference to the ``Freya`` package, which brings in all of the dependencies needed for a common Freya application.
+
+Code
+----
+
+At this point you should have a basic Freya application. :file:`Api.fs` starts off by opening some common namespaces that are needed to access the major components of the Freya stack.
+
+.. code-block:: fsharp
+
+   open Freya.Core
+   open Freya.Machines.Http
+   open Freya.Types.Http
+   open Freya.Routers.Uri.Template
+
+Greeting
+^^^^^^^^
+
+The next section deals with the core ``freya`` construct, and is broken into three parts:
+
+.. code-block:: fsharp
+
+    let routeName_ = Route.atom_ "name"
+
+    let name = freya {
+      let! nameO = Freya.Optic.get routeName_
+
+      match nameO with
+      | Some name -> return name
+      | None -> return "World"
+    }
+
+    let sayHello = freya {
+      let! name = name
+
+      return Represent.text (sprintf "Hello, %s!" name)
+    }
+
+The first line defines a constant lens that inspects the route of the request for a component named ``{name}`` and provides a view that makes it easy to access.
```

Comment:
It's actually a prism rather than a lens, if people are going to refer to Aether docs - it might be worth changing that (or just using "optic" if we're going to be less fussy).